tests/conftest.py

```diff
@@ -19,137 +22,321 @@ def app(request):
+        # so we don't end up with collisions
+        if user.email_address == "test+replaceme@digital.gov.uk":
+            user.email_address = "test+{}@digital.gov.uk".format(user.id)
+            db.session.add(user)
+            db.session.commit()
+
+        user_id = user.id
+
+    def teardown():
+        with app.app_context():
+            User.query.filter(User.id == user_id).delete()
+            db.session.commit()
+    request.addfinalizer(teardown)
+    return user_id
 
 
 # G8
 
 
 @pytest.fixture()
 def open_g8_framework(request, app):
-    _g8_framework(request, app, status='open')
+    _framework_fixture_inner(request, app, **dict(_g8_framework_defaults, status="open"))
 
 
 @pytest.fixture()
 def live_g8_framework(request, app):
-    _g8_framework(request, app, status='live')
+    _framework_fixture_inner(request, app, **dict(_g8_framework_defaults, status="live"))
+
+
+@pytest.fixture()
+def live_g8_framework_2_variations(request, app):
+    _framework_fixture_inner(request, app, **dict(
+        _g8_framework_defaults,
+        status="live",
+        framework_agreement_details=dict(_generic_framework_agreement_details, variations={
+            "banana": {
+                "createdAt": "2016-06-06T20:01:34.000000Z",
+            },
+            "toblerone": {
+                "createdAt": "2016-07-06T21:09:09.000000Z",
+            },
+        }),
+    ))
 
 
 # G6
 
 
 @pytest.fixture()
 def open_g6_framework(request, app):
-    _g6_framework(request, app, status='open')
+    _framework_fixture_inner(request, app, **dict(_g6_framework_defaults, status="open"))
 
 
 @pytest.fixture()
 def expired_g6_framework(request, app):
-    _g6_framework(request, app, status='expired')
+    _framework_fixture_inner(request, app, **dict(_g6_framework_defaults, status="expired"))
 
 
 # DOS
 
 
 @pytest.fixture()
 def open_dos_framework(request, app):
-    _dos_framework(request, app, status='open')
+    _framework_fixture_inner(request, app, **dict(_dos_framework_defaults, status="open"))
 
 
 @pytest.fixture()
 def live_dos_framework(request, app):
-    _dos_framework(request, app, status='live')
+    _framework_fixture_inner(request, app, **dict(_dos_framework_defaults, status="live"))
 
 
 @pytest.fixture()
 def expired_dos_framework(request, app):
-    _dos_framework(request, app, status='expired')
+    _framework_fixture_inner(request, app, **dict(_dos_framework_defaults, status="expired"))
+
+
+# Suppliers
+
+
+@pytest.fixture()
+def supplier_basic(request, app):
+    return _supplier_fixture_inner(request, app, supplier_kwargs={"supplier_id": 1})
+
+
+@pytest.fixture()
+def supplier_basic_alt(request, app):
+    return _supplier_fixture_inner(request, app, supplier_kwargs={"supplier_id": 2})
+
+
+# Users
+
+
+@pytest.fixture()
+def user_role_supplier(request, app, supplier_basic):
+    return _user_fixture_inner(request, app, user_kwargs={
+        "id": 1,
```

Comment:
Is there a reason to hardcode these IDs? I think there should be a way to let the DB create a new record and make the new ID available to the tests through the fixture value. This also makes me question the value of _alt version, but I'd have to think about how you could make the test require two user records at once.